Core/HttpHandlerProxy.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http.Headers;
+using System.Net.Http;
+using System.Net;
+using System.Security.Authentication;
+using System.Text;
+using Microsoft.Net.Http.Headers;
+using System.Threading.Tasks;
+using System.Threading;
+using Microsoft.Exchange.WebServices.Data;
+using Microsoft.Exchange.WebServices.NETStandard.Extensions;
+
+namespace Microsoft.Exchange.WebServices.NETStandard.Core
+{
+    internal class HttpHandlerProxy : DelegatingHandler
+    {
+        class SimpleHttpMessageHandlerProxy : DelegatingHandler
+        {
+            public SimpleHttpMessageHandlerProxy(HttpMessageHandler messageHandler)
+                : base(messageHandler) 
+            {
+            }
+
+            internal Task<HttpResponseMessage> SendImplAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                return base.SendAsync(request, cancellationToken);
+            }
+        }
+
+        private readonly string httpClientName;
+        private readonly HttpMessageHandler httpMessageHandler;
+        private readonly HttpClientHandler httpClientHandler;
+
+        /// <summary>
+        /// Gets or sets the cookie container.
+        /// </summary>
+        /// <value>The cookie container.</value>
+        public CookieContainer CookieContainer { get; set; }
+
+        /// <summary>
+        /// Gets or sets authentication information for the request.
+        /// </summary>
+        /// <returns>An <see cref="T:System.Net.ICredentials"/> that contains the authentication credentials associated with the request. The default is null.</returns>
+        public ICredentials Credentials { get; set; }
+
+        /// <summary>
+        /// Gets or sets proxy information for the request.
+        /// </summary>
+        public IWebProxy Proxy { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that indicates whether to send an authenticate header with the request.
+        /// </summary>
+        /// <returns>true to send a WWW-authenticate HTTP header with requests after authentication has taken place; otherwise, false. The default is false.</returns>
+        public bool PreAuthenticate { get; set; }

```

Comment:
Not used?

Reply:
Proxy, PreAuthenticate and UseDefaultCredentials properties are not used right now, I have added them for possible future use

Reply:
Comment out then?

Reply:
yes, it can be commented